lib/turn_utils.ex

```diff
@@ -0,0 +1,105 @@
+defmodule Membrane.TURN.Utils do
+  @moduledoc false
+
+  @numbers_and_letters Enum.concat([?0..?9, ?a..?z, ?A..?Z])
+
+  @spec generate_credentials(binary(), binary()) :: {binary(), binary()}
+  def generate_credentials(name, secret) do
+    duration =
+      DateTime.utc_now()
+      |> DateTime.to_unix()
+      |> tap(fn unix_timestamp -> unix_timestamp + 24 * 3600 end)
+
+    username = "#{duration}:#{name}"
+
+    password =
+      :crypto.mac(:hmac, :sha, secret, username)
+      |> Base.encode64()
+
+    {username, password}
+  end
+
+  @spec generate_secret() :: binary()
+  def generate_secret() do
+    symbols = '0123456789abcdef'
+
+    1..20
+    |> Enum.map(fn _i -> Enum.random(symbols) end)
+    |> to_string()
+  end
+
+  @spec start_integrated_turn(binary(), list()) :: {:ok, :inet.port_number(), pid()}
+  def start_integrated_turn(secret, opts \\ []),
+    do: :turn_starter.start(secret, opts)
+
+  @spec stop_integrated_turn(map()) :: :ok
+  def stop_integrated_turn(turn),
+    do: :turn_starter.stop(turn.server_addr, turn.server_port, turn.relay_type)
+
+  @spec generate_fake_ice_candidate({:inet.ip4_address(), :inet.port_number()}) :: binary()
+  def generate_fake_ice_candidate({ip, port}) do
+    ip
+    |> Tuple.to_list()
+    |> Enum.join(".")
+    |> then(&"a=candidate:1 1 UDP 2015363327 #{&1} #{port} typ host")
```

Comment:
What is this number `2015363327`? Is it priority? If so can we write why we chose such value?

Reply:
Yes, it is priority. Choose of `2015363327` was quite arbitrary ;)